### ISSUE TITLE
Failure after loading checkpoint shards.

### ISSUE THREAD

**Issue by ftgreat** [facebookresearch/metaseq#655]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
Use metaseq-train script, when training finished, checkpoint shards are generated properly.

But continual training failure happens after loading checkpoint shards. 

Could you please give me some advice about whether improper usage? @suchenzang 

#### Code

The initial training shell and reloading shell (including all flags) are the same as follows.

# two nodes and one gpu per node

metaseq-train \
  --task streaming_language_modeling \
  data-bin/pile-00 \
  --vocab-filename data-bin/pile-00/gpt2-vocab.json \
  --merges-filename data-bin/pile-00/gpt2-merges.txt \
  --criterion cross_entropy \
  --batch-size 1 \
  --save-dir checkpoints/lm_transformer_pile-00 \
  --arch transformer_lm --share-decoder-input-output-embed \
  --dropout 0.1 \
  --optimizer adam --weight-decay 0.01 --clip-norm 0.0 \
  --lr 0.0005 --lr-scheduler inverse_sqrt --warmup-updates 1 --warmup-init-lr 1e-07 \
  --tokens-per-sample 256 --sample-break-mode none \
  --fp16 \
  --bf16 \
  --dataset-impl lazy \
  --num-workers 4 \
  --decoder-learned-pos \
  --log-format json \
  --max-epoch 1 \
  --arch transformer_lm \
  --distributed-init-method tcp://$MASTER_ADDR:$MASTER_PORT \
  --distributed-world-size 2 \
  --ddp-backend fully_sharded \
  --checkpoint-shard-count 2 \
  --use-sharded-state \
  --log-interval 1 \
  --distributed-rank $RANK

   
Checkpoint shards are generated and i rename them to last version for reloading.
![image](https://user-images.githubusercontent.com/10432020/220143390-aa124f88-35d4-4283-9680-790c6a3efa0d.png)

Reload shards seem properly.
![image](https://user-images.githubusercontent.com/10432020/220143977-90b3a7ee-41f9-44d0-860c-530c67cfdd83.png)

But when start to training, failure happened.
![image](https://user-images.githubusercontent.com/10432020/220144247-f4a74939-6fb7-4dd3-8170-98b1b9ce3a07.png)

#### What's your environment?

 - metaseq Version (e.g., 1.0 or master):
 - PyTorch Version (e.g., 1.0)
 - OS (e.g., Linux):
 - How you installed metaseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:
